src/components/Modals/SaveCourseModal.vue

```diff
@@ -38,30 +43,52 @@
 <script lang="ts">
 import { defineComponent } from 'vue';
 import TeleportModal from '@/components/Modals/TeleportModal.vue';
+import store from '@/store';
 
 export default defineComponent({
   components: { TeleportModal },
   props: {
     courseCode: { type: String, required: true },
-    collection: { type: String, default: 'No collections added yet' },
-    isdefaultCollection: { type: Boolean, default: true },
+  },
+  data() {
+    return {
+      checkedCollections: [] as string[], // New data property to manage checked state
+    };
+  },
+  computed: {
+    isDefaultCollection() {
+      const collections = store.state.collections.map(collection => collection.name);
+      return collections.length === 0;
+    },
+    collections() {
+      const collections = store.state.collections.map(collection => collection.name);
+      return collections.length === 0 ? ['No collections added yet'] : collections;
```

Comment:
Clean way of doing things, nice approach! Took a minute to understand so perhaps a one-line piece of documentation could be useful, but definitely the most straightforward approach — doesn't really on any extra attributes or anything.